src/xwayland/XWM.cpp

```diff
@@ -691,6 +691,8 @@ int CXWM::onEvent(int fd, uint32_t mask) {
         Debug::log(CRIT, "XWayland has yeeten the xwm off?!");
         g_pXWayland->pWM.reset();
         g_pXWayland->pServer.reset();
+        // Attempt to create fresh instance
+        g_pXWayland = std::make_unique<CXWayland>(true);
```

Comment:
wont this already happen elsewhere?

Reply:
I don't think so as every time my XWayland crashes (a separate thing I haven't reasoned out yet) it requires disabling and re-enabling before it works again. Programs instead just pop up as if XWayland were disabled.

Reply:
If you have an area where you think this should be put instead I'm down take a look into it

Reply:
ah.. right. I see. This should probably be done in a `doLater`, though. Making it instantly here would be an awkward codepath.

Reply:
Good point, newest commit should resolve that